himbaechel/uarch/gatemate/pack.cc

```diff
@@ -26,196 +26,28 @@
 
 NEXTPNR_NAMESPACE_BEGIN
 
-void GateMatePacker::flush_cells()
+void GateMatePacker::flush_cells(bool unbind)
 {
     for (auto pcell : packed_cells) {
         for (auto &port : ctx->cells[pcell]->ports) {
             ctx->cells[pcell]->disconnectPort(port.first);
         }
+        if (unbind)
```

Comment:
is there any reason to make `unbind` optional? I would just make this conditional on `ctx->cells[pcell]->bel != BelId()`, erasing a cell without unbinding it is always going to be a problem...